From b2ac2eb66cbcb66ea111b2bb20ec7a92c6d5434a Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Tommi=20My=C3=B6h=C3=A4nen?= <ext-tommi.1.myohanen@nokia.com>
Date: Fri, 31 Jul 2009 11:27:03 +0300
Subject: [PATCH] outputselector: check for pending srcpad in _get_property()

If there is a pending srcpad, return it instead of active srcpad
in gst_output_selector_get_property() function.
---
 gst/selector/gstoutputselector.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/gst/selector/gstoutputselector.c b/gst/selector/gstoutputselector.c
index 15c0a72..f5c89ee 100644
--- a/gst/selector/gstoutputselector.c
+++ b/gst/selector/gstoutputselector.c
@@ -238,7 +238,8 @@ gst_output_selector_get_property (GObject * object, guint prop_id,
   switch (prop_id) {
     case PROP_ACTIVE_PAD:
       GST_OBJECT_LOCK (object);
-      g_value_set_object (value, sel->active_srcpad);
+      g_value_set_object (value,
+          sel->pending_srcpad ? sel->pending_srcpad : sel->active_srcpad);
       GST_OBJECT_UNLOCK (object);
       break;
     case PROP_RESEND_LATEST:{
